airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/assets/registry.py

```diff
@@ -89,12 +89,18 @@ def apply_release_candidates(
     latest_registry_entry: dict,
     release_candidate_registry_entry: PolymorphicRegistryEntry,
 ) -> dict:
-    updated_registry_entry = copy.deepcopy(latest_registry_entry)
-    updated_registry_entry.setdefault("releases", {})
-    updated_registry_entry["releases"]["releaseCandidates"] = {
-        release_candidate_registry_entry.dockerImageTag: to_json_sanitized_dict(release_candidate_registry_entry)
-    }
-    return updated_registry_entry
+    # Ensure that the release candidate is newer than the latest registry entry
+    if semver.Version.parse(release_candidate_registry_entry.dockerImageTag) > semver.Version.parse(
+        latest_registry_entry["dockerImageTag"]
+    ):
+        updated_registry_entry["releases"]["releaseCandidates"] = {
+            release_candidate_registry_entry.dockerImageTag: to_json_sanitized_dict(release_candidate_registry_entry)
+        }
+        updated_registry_entry = copy.deepcopy(latest_registry_entry)
+        updated_registry_entry.setdefault("releases", {})
```

Comment:
Should this go above the ` updated_registry_entry["releases"]["releaseCandidates"]` line?